tools/loader/ImageClassifier.cpp

```diff
@@ -459,11 +466,31 @@ static void runInference(runtime::HostManager *hostManager, std::string name,
                          std::unique_ptr<ExecutionContext> batch,
                          std::promise<void> &runPromise,
                          std::atomic<unsigned> &inflight,
-                         std::atomic<int> &dispatched, unsigned warmUp) {
+                         std::atomic<int> &dispatched, unsigned warmUp,
```

Comment:
Nit: The number of parameters for this function slowly becomes unwieldy and I have the feeling we may want to add more parameters in the future. May be some kind of a compound data structure should be used instead to pass the parameters? Or maybe we should introduce a proper class encapsulating all this data and make this function a method in this class?

Reply:
Perhaps, but converting to a class just moves the parameter list to a constructor so I'm not sure how much cleaner it would end up being. How about when there's time to add yet more parameters a refactoring could be in place?

Reply:
Fair enough ;-)